app/server/models/credit_transfer.py

```diff
@@ -172,19 +175,77 @@ def resolve_as_completed(self, existing_blockchain_txn=None):
             if self.recipient_user and self.recipient_user.transfer_card:
                 self.recipient_user.transfer_card.update_transfer_card()
 
-        if self.transfer_type == TransferTypeEnum.DEPOSIT and self.fiat_ramp:
+        if self.fiat_ramp and self.transfer_type in [TransferTypeEnum.DEPOSIT, TransferTypeEnum.WITHDRAWAL]:
             self.fiat_ramp.resolve_as_completed()
 
         if not existing_blockchain_txn:
             self.send_blockchain_payload_to_worker()
 
     def resolve_as_rejected(self, message=None):
+        if self.fiat_ramp and self.transfer_type in [TransferTypeEnum.DEPOSIT, TransferTypeEnum.WITHDRAWAL]:
+            self.fiat_ramp.resolve_as_rejected()
+
         self.resolved_date = datetime.datetime.utcnow()
         self.transfer_status = TransferStatusEnum.REJECTED
 
         if message:
             self.resolution_message = message
 
+    def get_transfer_limits(self):
+        relevant_limits = []
+        for limit in LIMITS:
+            applied = limit.application_filter(self)
+            if applied and str(self.transfer_type) in limit.applied_to_transfer_types:
+                relevant_limits.append(limit)
+
+        return relevant_limits
+
+    def check_sender_transfer_limits(self):
+        if self.sender_user is None:
+            # skip if an exchange
+            return
+
+        relevant_transfer_limits = self.get_transfer_limits()
+
+        for limit in relevant_transfer_limits:
+            filter_before = datetime.datetime.today() - datetime.timedelta(days=limit.time_period_days)
+
+            if limit.transfer_count is not None:
+                # GE Limits
+                transaction_count = db.session.query(func.count(CreditTransfer.id).label('count'))\
+                    .filter(CreditTransfer.created >= filter_before)\
+                    .filter(CreditTransfer.transfer_type == self.transfer_type) \
+                    .filter(CreditTransfer.sender_user == self.sender_user) \
+                    .execution_options(show_all=True).first().count
+
+                if (transaction_count or 0) > limit.transfer_count:
+                    message = 'Account Limit "{}" reached. Allowed {} transaction per {} days'\
+                        .format(limit.name, limit.transfer_count, limit.time_period_days)
+                    self.resolve_as_rejected(message=message)
+                    raise AccountLimitError(message)
+
+                amount_avail = limit.transfer_balance_percentage*self.sender_transfer_account.balance
+                if self.transfer_amount > amount_avail:
+                    message = 'Account Limit "{}" reached. {} available'.format(limit.name, amount_avail)
+                    self.resolve_as_rejected(message=message)
+                    raise AccountLimitError(message)
+
+            else:
```

Comment:
instead of an else that distinguishes GE vs sempo, we should keep them generic. if there's a `total_amount` limit, then run this condition. then we could have any combination of different limits

Reply:
Hm ok. will take a look and ship update

Reply:
ok updated